### PR TITLE
feat: add with_generators

### PR DESCRIPTION
**Pull request by JP-Ellis**
Add FFI to add generators without requiring the use of integration JSON.

### CODE REVIEW

**Comment by JP-Ellis** [comment]
Before merging, I just want to add a few tests first; though I can confirm that it works locally in Pact Python.

File: rust/pact_ffi/src/mock_server/handles.rs [hunk 1]
```diff
@@ -2119,6 +2119,93 @@ ffi_fn!{
   }
 }
 
+ffi_fn!{
+  /// Add generators to the interaction.
+  ///
+  /// * `interaction` - Interaction handle to set the generators for.
+  /// * `part` - Request or response part (if applicable).
+  /// * `rules` - JSON string of the generators to add to the interaction.
```
Comment (rholshausen) [inline rust/pact_ffi/src/mock_server/handles.rs:2127]: Comment doesn't match the parameter names

File: rust/pact_ffi/src/mock_server/handles.rs [hunk 2]
```diff
@@ -2119,6 +2119,93 @@ ffi_fn!{
...
+  /// # Safety
+  ///
+  /// The rules parameter must be a valid pointer to a NULL terminated UTF-8
+  /// string.
+  fn pactffi_with_generators(
+    interaction: InteractionHandle,
+    part: InteractionPart,
+    generators: *const c_char
+  ) -> bool {
+    let generators = match convert_cstr("rules", generators) {
```
Comment (rholshausen) [inline rust/pact_ffi/src/mock_server/handles.rs:2145]: should be generators not rules

**Comment by JP-Ellis** [comment]
Fixed the typos, and added a test. Ready for review :)

**Pull request merged**